backfill_corrections/delphiBackfillCorrection/R/model.R

```diff
@@ -262,3 +267,63 @@ generate_filename <- function(indicator, signal,
   )
   return(filename)
 }
+
+#' Get date range of data to use for training models
+#'
+#' Calculate training start and end dates based on user settings.
+#' `training_start_date` is the minimum allowed target date when selecting
+#' training data to use. `training_end_date` is the maximum allowed target
+#' date and maximum allowed issue date.
+#'
+#' Cases:
+#'   1. We are training new models.
+#'   2. We are not training new models and cached models exist.
+#'   3. We are not training new models and cached models don't exist.
+#'
+#' Sometimes we want to allow the user to specify an end date in
+#' params that overrides the automatically-generated end date. This is
+#' only relevant when the user requests to train new models.
+#'
+#' @template params-template
+get_training_date_range <- function(params) {
```

Comment:
Together with the problem that I mentioned in PR #1712, I would suggest that we put this function under `utils.R` or `io.R`.

Reply:
NVM. But remember that we don't have unit tests for `get_training_date_range` currently.

Reply:
I'll make an issue for that and aim to address soon.